app/src/main/java/com/example/voyagerx/data/LoginRepository.kt

```diff
@@ -0,0 +1,46 @@
+package com.example.voyagerx.data
+
+import com.example.voyagerx.data.model.LoggedInUser
+
+/**
+ * Class that requests authentication and user information from the remote data source and
+ * maintains an in-memory cache of login status and user credentials information.
+ */
+
+class LoginRepository(val dataSource: LoginDataSource) {
+
+    // in-memory cache of the loggedInUser object
+    var user: LoggedInUser? = null
+        private set
```

Comment:
How are we getting the current logged in user?

Reply:
That's going to be saved in sharedpreferences. I haven't implemented it yet but will soon.